tests/test_hive.py

```diff
@@ -0,0 +1,190 @@
+        local_path = os.path.join(temp_dir, file_name)
+        with open(local_path, 'w') as fp:
+            fp.write(template_contents.replace("{{HOSTNAME}}", hive_agent_hostname))
+        url = spark_utils.upload_file(local_path)
+    return url
+
+
+@pytest.fixture(scope='module')
+def hadoop_setup(kerberos_setup):
+    try:
+        # TODO: replace with Marathon readiness check in kdc app?
+        #log.info("Waiting for kdc.marathon.mesos to resolve ...")
+        #time.sleep(10)
+
+        # Run the Hive Docker image in Marathon
+        curr_dir = os.path.dirname(os.path.realpath(__file__))
+        app_def_path = "{}/resources/hadoop-hive-kerberos.json".format(curr_dir)
+        with open(app_def_path) as f:
+            hive_app_def = json.load(f)
+        hive_app_def["id"] = HIVE_APP_ID
+        hive_app_def["constraints"] = [["hostname", "IS", kerberos_setup.hive_agent_ip]]
+        sdk_marathon.install_app(hive_app_def)
+        # Wait for Sentry and Hive to be ready
+        # TODO: replace with Marathon health check
+        log.info("Sleeping 120s ...")
+        time.sleep(120)
+
+        # HDFS client configuration
+        core_site_url = _upload_hadoop_config("core-site.xml", kerberos_setup.hive_agent_hostname)
+        _upload_hadoop_config("hdfs-site.xml", kerberos_setup.hive_agent_hostname)
+        hive_config_url = _upload_hadoop_config("hive-site.xml", kerberos_setup.hive_agent_hostname)
+        hdfs_base_url = os.path.dirname(core_site_url)
+        yield HadoopSetup(kerberos_setup, hdfs_base_url, hive_config_url)
+
+    finally:
+        sdk_marathon.destroy_app(HIVE_APP_ID)
+
+
+@pytest.fixture(scope='module')
+def configure_security_spark():
+    yield from spark_utils.spark_security_session()
+
+
+@pytest.fixture(scope='module', autouse=True)
+def setup_spark(kerberos_setup, hadoop_setup, configure_security_spark):
+    try:
+        spark_utils.upload_dcos_test_jar()
+        additional_options = {
+            "hdfs": {
+                "config-url": hadoop_setup.hdfs_base_url
+            },
+            "security": {
+                "kerberos": {
+                    "enabled": True,
+                    "realm": sdk_auth.REALM,
+                    "kdc": {
+                        "hostname": kerberos_setup.kerberos_env.get_host(),
+                        "port": int(kerberos_setup.kerberos_env.get_port())
+                    }
+                }
+            }
+        }
+        spark_utils.require_spark(additional_options=additional_options)
+        yield
+    finally:
+        spark_utils.teardown_spark()
+
+
+def _grant_hive_privileges():
+    rc, _, err = sdk_cmd.marathon_task_exec(HIVE_APP_ID, "/etc/grant-hive-privileges.sh")
+    if rc != 0:
+        raise Exception("Hive script failed with code {}: {}".format(rc, err))
+
+
+@pytest.mark.xfail(sdk_utils.is_strict_mode(),
+    reason="hadoop_setup requires root access; won't work in strict mode")
```

Comment:
I don't quite think that this reasoning is valid? Which part of the application requires `root` access? Can we not then run the application as the `root` user (giving Marathon the required permissions to launch it successfully).